### PR TITLE
adding metadata lambda function to repository for version control

### PR DESCRIPTION
**Pull request by PierceLovesee**
Lambda function for extracting metadata from all videos in a directory on S3 and writing specified metadata fields (right now, fps and bitrate) to dynamoDB.

Depends on execution from invoke_lambda_async.py.

### CODE REVIEW

File: benchmarking/video_quality/lambda_metadata.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import subprocess
```
Comment (vinayakSharm) [inline benchmarking/video_quality/lambda_metadata.py:1]: module docstring

File: benchmarking/video_quality/lambda_metadata.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+s3_client = boto3.client("s3")
+dynamodb = boto3.resource("dynamodb")
+
+def lambda_handler(event, context):
+
+    """
+    Lambda function to process video files, extract metadata, and upload to DynamoDB.
+
+    Args:
```
Comment (vinayakSharm) [inline benchmarking/video_quality/lambda_metadata.py:14]: add the " -------" underneath the args and returns for standardized format

File: benchmarking/video_quality/lambda_metadata.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    s3_keys = list_object_keys(bucket_name, folder_path)
+
+    # define the dynamoDB table
+    table_name = dynamodb_table
+    table = dynamodb.Table(table_name)
+
+    for key in s3_keys:
+    
+        # define local file name and download video to lambda
+        local_filename = "/tmp/" + os.path.basename(key)
```
Comment (vinayakSharm) [inline benchmarking/video_quality/lambda_metadata.py:37]: nice --> like the .basename

File: benchmarking/video_quality/lambda_metadata.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        except Exception as e:
+            print("Error uploading key:", key, e)
+
+
+def list_object_keys(bucket_name, folder_path):
+
+    """
+    List object keys within a specified S3 bucket and folder path.
+
+    Args:
```
Comment (vinayakSharm) [inline benchmarking/video_quality/lambda_metadata.py:69]: docstring stuff : "-----" underneath the args and returns

**Review by vinayakSharm — CHANGES_REQUESTED** [review]
small docstring updates. 

question on the validity of list_object_keys if video not in mp4 encoding?

**Review by hamzakhokhar — APPROVED** [review]

**Review by vinayakSharm — APPROVED** [review]
good.

**Pull request merged**